core/arch/opnd_shared.c

```diff
@@ -2118,6 +2125,15 @@ reg_get_size(reg_id_t reg)
         return OPSZ_16;
     if (reg >= REG_START_YMM && reg <= REG_STOP_YMM)
         return OPSZ_32;
+    if (reg >= REG_START_ZMM && reg <= REG_STOP_ZMM)
+        return OPSZ_64;
+    if (reg >= REG_START_OPMASK && reg <= REG_STOP_OPMASK) {
+        /* The default is 16 bits wide. The register may be up to 64 bits wide with
+         * the AVX-512BW extension, which depends on the processor. The number of
+         * bits actually used depends on the vector type of the instruction.
+         */
+        return OPSZ_64;
```

Comment:
It seems like this should be a new variable size class and not this inaccurate OPSZ_64.

Reply:
I have added a XXX comment. It can by dynamic once we have a proc switch for the AVX512BW feature.

Reply:
First, 64 bits is OPSZ_8, not OPSZ_64.  Second, I am suggesting a variable-size class.  Decoding is sometimes done statically with no context so we would have an unresolved size in such a case.  This would be sthg like OPSZ_2_bw8.  I'm making up the "bw": I don't know what controls this well enough to know what the name should be.

Reply:
Ok, thanks. I meant OPSZ_8 indeed. What I wanted eventually was
return AVX512BW_ENABLED() ? OPSZ_8 ? OPSZ_2;

Regarding second: I'll look at this (AFK), maybe I misunderstood the meaning of the OPSZ in this case.

Reply:
The other way would be to give the registers different enum values for the different sizes, like RAX vs EAX vs AX vs AL.  I'm not sure I understand these mask registers well enough: are they 64 bits and typically only the bottom 16 bits matter?  What other sizes occur?  Just 16 or 64?  Is this difference a static property or is there some control bit in an opcode which can change it?

Reply:
The AVX-512 vector lanes can be multiplexed dependent on the instruction. Initially they had single and double precision, dword and qword. The mask in this case 8-16 bits (above should actually be OPSZ_1). So the physical register is 16 bits but some instructions only read 8 bits. There are also instructions that use the mask registers as a scalar like kmov [m], k1 etc., e.g. kmov for 8 or 16 bits. 

AVX512BW adds the 16 bit and 8 bit integer lanes, they call it neural extensions, because they market it to be used in machine learning They can do FMA with 64 bytes etc. In any case, the mask in this case is 32 - 64 bits (masks up to 64 individual 8 bit lanes).

I assumed reg_get_size should return the size of the physical register, regardless of what different instructions read.

Reply:
Changed into OPSZ_8. Will be switched into the real physical size later. To my understanding, the function returns the physical size, not the instruction dependent size or how it's used.